contracts/interchain-token/InterchainToken.sol

```diff
@@ -17,21 +15,38 @@ import { Distributable } from '../utils/Distributable.sol';
  * @notice This contract implements a interchain token which extends InterchainToken functionality.
  * This contract also inherits Distributable and Implementation logic.
  */
-contract InterchainToken is BaseInterchainToken, ERC20Permit, Implementation, Distributable, IInterchainToken {
+contract InterchainToken is BaseInterchainToken, ERC20Permit, Distributable, IInterchainToken {
     using AddressBytes for bytes;
 
     string public name;
     string public symbol;
     uint8 public decimals;
     address internal tokenManager_;
 
-    bytes32 private constant CONTRACT_ID = keccak256('interchain-token');
+    // bytes32(uint256(keccak256('interchain-token-is-setup-slot')) - 1);
```

Comment:
Can we keep the `implementation()` method? Some explorers might not detect the minimal proxy

Reply:
no we cannot